perun-cli/removeUserExtSource

```diff
@@ -11,44 +11,59 @@ sub help {
 	Removes an external source from the user. User id and external login and external source id or name are required fields
 	------------------------------------
 	Available options:
-	--userId         | -u user id
-	--extSourceId    | -e external source id
-	--extSourceName  | -E external source name
-	--extSourceLogin | -l external login
-	--batch          | -b batch
-	--help           | -h prints this help
+	--userId          | -u user id
+	--extSourceId     | -e external source id
+	--extSourceName   | -E external source name
+	--extSourceLogin  | -l external login
+	--userExtSourceId | -x user external source id
+	--forceRemove     | -f remove persistent external source too
+	--batch           | -b batch
+	--help            | -h prints this help
 
 	};
 }
 
-my ($userId, $extSourceId, $extSourceName, $extSourceLogin, $batch);
+my ($userId, $extSourceId, $extSourceName, $extSourceLogin, $batch, $force,$userExtSourceId);
 GetOptions ("help|h"     => sub {
 		print help();
 		exit 0;
 	}, "userId|u=i"      => \$userId,
-	"extSourceId|e=i"    => \$extSourceId, "extSourceName|E=s" => \$extSourceName,
-	"extSourceLogin|l=s" => \$extSourceLogin, "batch|b" => \$batch) || die help();
+	"extSourceId|e=i"    => \$extSourceId, 
+	"extSourceName|E=s" => \$extSourceName,
+	"extSourceLogin|l=s" => \$extSourceLogin, 
+	"userExtSourceId|x=i" => \$userExtSourceId,
+	"forceRemove|f" => \$force,
+	"batch|b" => \$batch) || die help();
 
 # Check options
 unless (defined($userId)) { die "ERROR: userId is required \n";}
-unless (defined($extSourceId) or defined($extSourceName)) { die "ERROR: extSourceId or extSourceName is required \n";}
+unless (defined $userExtSourceId) {
+	if ((!defined($extSourceId) and !defined($extSourceName)) or !defined($extSourceLogin)) { die "ERROR: extSourceId or extSourceName and extSourceLogin is required \n";}
+}
 
 my $agent = Perun::Agent->new();
 my $usersAgent = $agent->getUsersAgent;
 my $extSourcesAgent = $agent->getExtSourcesAgent;
 
 my $extSource;
-if ($extSourceId) {
-	$extSource = $extSourcesAgent->getExtSourceById( id => $extSourceId );
-}
-if ($extSourceName) {
-	$extSource = $extSourcesAgent->getExtSourceByName( name => $extSourceName );
-	$extSourceId = $extSource->getId;
-}
+unless (defined $userExtSourceId) {
+	if ($extSourceId) {
+		$extSource = $extSourcesAgent->getExtSourceById( id => $extSourceId );
+	}
+	if ($extSourceName) {
```

Comment:
I would prefer ``elsif`` since if anybody passes both params (id and name), then name takes preference and it overrides extSourceId value. And such behavior is not documented in tool help. We could also die in else branch, but presence of one of params is checked before, so it's not necessary.